### ISSUE TITLE
ATInternet-Apple-SDK/watchOSTracker 2.15.1 on Watch OS 5

### ISSUE THREAD

**Issue by nimau** [at-internet/atinternet-apple-sdk#67]
Hi,
I'm using ATInternet using pod:

target 'WatchApp Extension' do
  platform :watchos, '5.0'
  use_frameworks!
  pod 'ATInternet-Apple-SDK/watchOSTracker'
end

The version used is the last one: 2.15.1

All is working fine on WatchOS 6, but on WatchOS 5 the app won't launch:

dyld: lazy symbol binding failed: Symbol not found: _WKExtensionMain

Removing the pod 'ATInternet-Apple-SDK/watchOSTracker' fix the issue and the app runs again on WatchOS 5.

Is this a known issue ?

Regards.

**Comment by nsagnett** [comment]
Hi, 

What kind of problem do you have? And when do it occurs ? Runtime ? Build ? 

Can you share some more input to help you please ?

**Comment by nimau** [comment]
Hi,

Unfortunately, I don't really have more input to provide.

When I add pod 'ATInternet-Apple-SDK/watchOSTracker' to the podfile, the app crash on launch with the following error in Xcode:

dyld: lazy symbol binding failed: Symbol not found: _WKExtensionMain
  Referenced from: <path>/WatchApp.app/PlugIns/WatchApp Extension.appex/WatchApp Extension
  Expected in: /Library/Developer/CoreSimulator/Profiles/Runtimes/watchOS 5.2.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/WatchKit.framework/WatchKit

dyld: Symbol not found: _WKExtensionMain
  Referenced from: <path>/WatchApp.app/PlugIns/WatchApp Extension.appex/WatchApp Extension
  Expected in: /Library/Developer/CoreSimulator/Profiles/Runtimes/watchOS 5.2.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/WatchKit.framework/WatchKit

It crashes both on simulator and device when running watchOS 5.2

On WatchOS 6 it's working fine.

Also, on watchOS 5.2, if I remove the pod, the app runs fine.

Hope it helps.

**Comment by nsagnett** [comment]
Can you share a sample of your project to be in same conditions, please ?

**Comment by nimau** [comment]
Well, I have the same issue when a create a new project.
With Xcode 11, I have created a new project "iOS App with WatchApp". With target set to:
iOS 11 and WatchOS 5.
I've added the following podfile:
```
target 'test-atinternet' do
  platform :ios, '11.0'
  use_frameworks!

  pod 'ATInternet-Apple-SDK/Tracker'
end

target 'test-atinternet WatchKit Extension' do
  platform :watchos, '5.0'
  use_frameworks!
  
  pod 'ATInternet-Apple-SDK/watchOSTracker'
end
```

When I try to deploy the app on a watchOS 5 simulator, I have the following error:
```
dyld: lazy symbol binding failed: Symbol not found: _WKExtensionMain
  Referenced from: /Users/nicolas/Library/Developer/CoreSimulator/Devices/5352E7B7-706F-4703-A1A4-A502A556E2EA/data/Containers/Bundle/Application/A338943D-011D-4C04-B4BE-1D9EEF3FF544/test-atinternet WatchKit App.app/PlugIns/test-atinternet WatchKit Extension.appex/test-atinternet WatchKit Extension
  Expected in: /Library/Developer/CoreSimulator/Profiles/Runtimes/watchOS 5.2.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/WatchKit.framework/WatchKit

dyld: Symbol not found: _WKExtensionMain
  Referenced from: /Users/nicolas/Library/Developer/CoreSimulator/Devices/5352E7B7-706F-4703-A1A4-A502A556E2EA/data/Containers/Bundle/Application/A338943D-011D-4C04-B4BE-1D9EEF3FF544/test-atinternet WatchKit App.app/PlugIns/test-atinternet WatchKit Extension.appex/test-atinternet WatchKit Extension
  Expected in: /Library/Developer/CoreSimulator/Profiles/Runtimes/watchOS 5.2.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks/WatchKit.framework/WatchKit

```

**Comment by nsagnett** [comment]
I'm afraid but i can't reproduce.

I created a project like yourself, set deployment target to iOS 11.0 and WatchOS 5.0
Run on simulators : 
- iOS : iPhone Pro 13.0
- watchOS : Apple watch series 2 - 5.0

And I can't run the project, i have the following error
```
Details

This app could not be installed at this time.
Domain: IXUserPresentableErrorDomain
Code: 1
Failure Reason: WatchKit 1.0 apps are no longer installable on this watchOS version.
Recovery Suggestion: WatchKit 1.0 apps are no longer installable on this watchOS version.
--
WatchKit 1.0 apps are no longer installable on this watchOS version.
Domain: MIInstallerErrorDomain
Code: 133
User Info: {
    FunctionName = "-[MIInstallableBundle performVerificationWithError:]";
    LegacyErrorString = UnsupportedWatchKitVersion;
    SourceFileLine = 1357;
}
--
```

**Comment by nimau** [comment]
You probably don't have set watchOS 5 as target on both watch targets (Watch app and Watch extension)

**Comment by nsagnett** [comment]
When I'm in same configuration , the only one error is `dyld: Library not loaded: /System/Library/Frameworks/SwiftUI.framework/SwiftUI`

**Comment by nimau** [comment]
Yes, SwiftUI is only available for iOS 13 and watchOS 6.
You need to a project using Storyboard instead of SwiftUI. 
On the newly created project, set deployment targets (iOS 12 and watchOS 5).
In SceneDelegate.swift file, add @available(iOS 13, *) before each methods. (or remove the file)
Also in AppDelegate.swift, add @available(iOS 13, *) before UISceneSession methods (or remove the methods)
Finally add the following property in AppDelegate:
var window: UIWindow?

Add a simulator running on iOS 12, linked to a watchOS 5 simulator.

Ensure the project runs on watchOS 5 simulator.

Add your pod file.

**Comment by nsagnett** [comment]
After some tests , I think is Cocoapods issue because when I add frameworks manually as optional, the project runs correctly

**Comment by nimau** [comment]
Which frameworks are you setting as optional ?

**Comment by nsagnett** [comment]
AT-Internet frameworks from our delivery interface (https://apps.atinternet-solutions.com/TagComposer/#/)

**Comment by nsagnett** [comment]
I just pushed a new branch (fix/pod), where I clean parts of project, using canImport where it's needed to delete frameworks entries in podspec.
Can you try with this version ? 

Thanks in advance for your feedback

**Comment by nimau** [comment]
Hi, it seems to work. On both watch OS 5 and 6 simulators.
I can't try on watch os 5 device but it should be OK.
Thanks.

**Comment by nsagnett** [comment]
It's fixed in 2.15.3 release

Thanks for your feedback
